etl-jobs/src/main/scala/org/learner/jobs/UserLookupMigration.scala

```diff
@@ -0,0 +1,128 @@
+    val userLookupSchema = Encoders.product[UserLookup].schema;
+    val userLookupDF =  spark.read.format("org.apache.spark.sql.cassandra").schema(userLookupSchema).option("keyspace", "sunbird").option("table", "user_lookup").load();
+    
+
+    //Update value as email@orgId, phone@orgId, username@orgId, externalid@orgId
+
+    val stateUserLookupDF = filteredStateUserData.select(col("externalId"),col("userid"),col("provider")).withColumn("type",lit("externalid")).withColumn("value",concat(col("externalid"),lit("@"),col("provider"))).withColumn("userid",col("userid")).select(col("type"),col("value"),col("userid"));
+
+    val emailUserLookupDF = filteredUserData.select(col("email"),col("id")).withColumn("type",lit("email")).withColumn("value",concat(col("email"),lit("@"),translationMap(col("id")))).withColumnRenamed("id","userid").select(col("type"),col("value"),col("userid"));
+
+    val phoneUserLookupDF = filteredUserData.select(col("phone"),col("id")).withColumn("type",lit("phone")).withColumn("value",concat(col("phone"),lit("@"),translationMap(col("id")))).withColumnRenamed("id","userid").select(col("type"),col("value"),col("userid"));
+
+    val usernameUserLookupDF = filteredUserData.select(col("username"),col("id")).withColumn("type",lit("username")).withColumn("value",concat(col("username"),lit("@"),translationMap(col("id")))).withColumnRenamed("id","userid").select(col("type"),col("value"),col("userid"));
+
+   
+    val emailUserLookupDFwithoutNull = emailUserLookupDF.where(col("value").isNotNull)
+    val phoneUserLookupDFwithoutNull = phoneUserLookupDF.where(col("value").isNotNull)
+    val usernameUserLookupDFwithoutNull = usernameUserLookupDF.where(col("value").isNotNull)
+
+    println("Email Record Count:"+emailUserLookupDFwithoutNull.count())
+    println("Phone Record Count:"+phoneUserLookupDFwithoutNull.count())
+    println("Username Record Count:"+usernameUserLookupDFwithoutNull.count())
+    println("Username Record Count:"+stateUserLookupDF.count())
+
+    //Save reecords to the user_lookup table
+
+    stateUserLookupDF.write.format("org.apache.spark.sql.cassandra").option("keyspace", "sunbird").option("table", "user_lookup").mode(SaveMode.Append).save();
+    emailUserLookupDFwithoutNull.write.format("org.apache.spark.sql.cassandra").option("keyspace", "sunbird").option("table", "user_lookup").mode(SaveMode.Append).save();
+    phoneUserLookupDFwithoutNull.write.format("org.apache.spark.sql.cassandra").option("keyspace", "sunbird").option("table", "user_lookup").mode(SaveMode.Append).save();
+    usernameUserLookupDFwithoutNull.write.format("org.apache.spark.sql.cassandra").option("keyspace", "sunbird").option("table", "user_lookup").mode(SaveMode.Append).save();
+
+    val newTableRecords = spark.read.format("org.apache.spark.sql.cassandra").schema(userLookupschema).option("keyspace", "sunbird").option("table", "user_lookup").load().count();
+    println("user_lookup count post migration: " + newTableRecords);
```

Comment:
After doing this, I'd like some extra metrics as well to be done - all in the user table (not in lookup).

How many duplicate? If possible do a join on these dataframes and output their uuids.
1. usernames 
2. emails
3. phone